gnome-image-installer/util/gis-dmi.c

```diff
@@ -0,0 +1,112 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *
+ */
+
+#include "gis-dmi.h"
+#include "config.h"
+
+#include <gio/gio.h>
+
+#define SYS_VENDOR "/sys/class/dmi/id/sys_vendor"
+#define SYS_PRODUCT "/sys/class/dmi/id/product_name"
+
+/**
+ * gis_dmi_sanitize_string:
+ * @string: (transfer none): an arbitrary bytestring
```

Comment:
Well, in theory it could have embedded NULs, but we already assume in many places that there are no NULs in the DMI data. The type is courtesy of `g_file_get_contents()` which fetches raw bytes but returns them as `gchar *`. I don't think introducing new types (and casts) will improve clarity, particularly given the function is documented to remove bytes (not characters) outside the ASCII printable range, but I'll try to reword the docs to make this clearer.